core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/request/BaseResourceCreateRequest.java

```diff
@@ -27,73 +29,66 @@
 
 	protected static final long serialVersionUID = 1L;
 
-	// the new codesystem's ID, if not specified, it will be auto-generated
+	// the new ID, if not specified, it will be auto-generated
 	@NotEmpty
-	protected String id;
+	String id;
 	
 	@NotEmpty
-	protected String url;
+	String url;
 	
 	@NotEmpty
-	protected String title;
+	String title;
 	
-	protected String language;
-	protected String description;
-	protected String status;
-	protected String copyright;
-	protected String owner;
-	protected String contact;
-	protected String usage;
-	protected String purpose;
+	String language;
+	String description;
+	String status;
+	String copyright;
+	String owner;
+	String contact;
+	String usage;
+	String purpose;
 	
 	@NotEmpty
-	protected String bundleId;
-
-	public final void setId(String id) {
-		this.id = id;
-	}
-
-	public final void setUrl(String url) {
-		this.url = url;
-	}
-
-	public final void setTitle(String title) {
-		this.title = title;
-	}
-
-	public final void setLanguage(String language) {
-		this.language = language;
-	}
-
-	public final void setDescription(String description) {
-		this.description = description;
-	}
-
-	public final void setStatus(String status) {
-		this.status = status;
-	}
-
-	public final void setCopyright(String copyright) {
-		this.copyright = copyright;
-	}
-
-	public final void setOwner(String owner) {
-		this.owner = owner;
-	}
-
-	public final void setContact(String contact) {
-		this.contact = contact;
+	String bundleId;
+	
+	@Override
+	public final String execute(TransactionContext context) {
+		executeAdditionalLogic(context);
+		context.add(createResourceDocument());
+		return id;
 	}
-
-	public final void setUsage(String usage) {
-		this.usage = usage;
+	
+	public final String getId() {
+		return id;
 	}
-
-	public final void setPurpose(String purpose) {
-		this.purpose = purpose;
+	
+	public final String getTitle() {
+		return title;
 	}
+	
+	/**
+	 * Subclasses may override this method to add extra logic to request execution.
+	 */
+	protected void executeAdditionalLogic(final TransactionContext context) { }
 
-	public final void setBundleId(String bundleId) {
-		this.bundleId = bundleId;
+	protected abstract ResourceDocument.Builder setSpecializedFields(final ResourceDocument.Builder builder);
```

Comment:
Another badly named method. How about
```suggestion
        TODO javadoc
	protected abstract ResourceDocument.Builder prepareResource(final ResourceDocument.Builder builder);
```